lib/rubocop/cop/ezcater/graph_ql_type_file_order.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Ezcater
+      class GraphQlTypeFileOrder < Cop
+        ORDERED_DSL_METHODS = %w(
+          extend
+          include
+          description
+          pundit_policy_class
+          pundit_role
+          argument
+          field
+          user_errors_field
+        ).freeze
```

Comment:
I know it's not mentioned in the Eng. Portal doc, but do we discourage use of [`ready?`](https://graphql-ruby.org/mutations/mutation_authorization.html)? If not, should we accommodate that in here or in the resolve methods? We do have active usages of `ready?` in a few ez-rails mutations and I'd hate for someone to edit those and end up with confusing output from this cop.

Reply:
That's a great callout.

Given that we use Pundit I can't think of a reason to use `ready?` for any new code.

Given the current usage in ez-rails I think it's reasonable to add this to the ordering for now. If we as a team agree that `ready?` should not be used we can add another rule for that.